### ISSUE TITLE
Dark High Contrast - text highlighted by ctrl-f is not legible

### ISSUE THREAD

**Issue by mdavis-xyz** [jupyter/notebook#7537]
## Description

When selecting the dark high contrast theme, and using ctrl-f to search for text, the highlighted matched text is not readable. Maybe change the font colour to something dark for text matching a search pattern?

![Image](https://github.com/user-attachments/assets/9c573174-2799-4ff5-879a-2cda30e617bd)

## Reproduce

1. Change theme to dark high contrast
2. Create a code cell, with contents

```
abc
abc
abc
```

4. ctrl-f to search for `ab`
5. See the `ab` part of each line highlighted

## Expected behavior

It should be clear which text is highlighted, but also the value of the highlighted text.

## Actual Behavior

The matched text is highlighted with a background colour equal to the font colour, so the text can't be read.

Ok, I searched for `ab` so I know the highlighted text will contain `ab`. That's what makes this such a negligible bug. But perhaps I mis-typed, and I'm about to replace all for the wrong search text. Seeing it in the code would be helpful.

## Context

- Operating System and version: Ubuntu 24.04 (server)
- Browser and version: Firefox 132.0.2 - inside Ubuntu Gnome with the OS dark theme turned on

- Jupyter Notebook version: 

```
$ jupyter --version
Selected Jupyter core packages...
IPython          : 8.29.0
ipykernel        : 6.29.5
ipywidgets       : 8.1.5
jupyter_client   : 8.6.3
jupyter_core     : 5.7.2
jupyter_server   : 2.14.2
jupyterlab       : 4.3.2
nbclient         : 0.10.0
nbconvert        : 7.16.4
nbformat         : 5.10.4
notebook         : not installed
qtconsole        : not installed
traitlets        : 5.14.3

```

**Comment by krassowski** [comment]
The styles for it are defined in [here](https://github.com/jupyterlab/jupyterlab/blob/6e1174e202b6bf5a7b725d59b5ddafc5f0afbed2/packages/codemirror/style/base.css#L70-L97). I think that for the fix we would need to override/adjust `--jp-search-selected-match-background-color` value for dark high contrast theme.

**Comment by krassowski** [comment]
Is setting the text color to black here appropriate?

**Comment by mdavis-xyz** [comment]
Yes, black should be fine, I think.

Thanks!
